src/Microsoft.DotNet.Tools.Compiler/Program.cs

```diff
@@ -226,14 +243,16 @@ private static bool Compile(ProjectContext context, string configuration, string
                 foreach (var projectDependency in Sort(projects))
                 {
                     // Skip compiling project dependencies since we've already figured out the build order
-                    var compileResult = Command.Create("dotnet-compile", 
+                    var compileResult = ResolveCommand("dotnet-compile", 
```

Comment:
I'll try and move a centralized safety checking mechanism and revert the in place ones (like ResolveCommand here)